Core/AVIDump.cpp

```diff
@@ -0,0 +1,360 @@
+		// Let's boil it down to how we need to interpret the bits.
+		int baseFmt = buf.GetFormat() & ~(GPU_DBG_FORMAT_REVERSE_FLAG | GPU_DBG_FORMAT_BRSWAP_FLAG);
+		bool rev = (buf.GetFormat() & GPU_DBG_FORMAT_REVERSE_FLAG) != 0;
+		bool brswap = (buf.GetFormat() & GPU_DBG_FORMAT_BRSWAP_FLAG) != 0;
+		bool flip = buf.GetFlipped();
+
+		temp = new u8[3 * w * h];
+
+		// This is pretty inefficient.
+		const u16 *buf16 = (const u16 *)buffer;
+		const u32 *buf32 = (const u32 *)buffer;
+		for (u32 y = 0; y < h; y++) {
+			for (u32 x = 0; x < w; x++) {
+				u8 *dst;
+				if (flip) {
+					dst = &temp[(h - y - 1) * w * 3 + x * 3];
+				}
+				else {
+					dst = &temp[y * w * 3 + x * 3];
+				}
+
+				u8 &r = brswap ? dst[2] : dst[0];
+				u8 &g = dst[1];
+				u8 &b = brswap ? dst[0] : dst[2];
+
+				u32 src;
+				switch (baseFmt) {
+				case GPU_DBG_FORMAT_565:
+					src = buf16[y * buf.GetStride() + x];
+					if (rev) {
+						src = bswap16(src);
+					}
+					r = Convert5To8((src >> 0) & 0x1F);
+					g = Convert6To8((src >> 5) & 0x3F);
+					b = Convert5To8((src >> 11) & 0x1F);
+					break;
+				case GPU_DBG_FORMAT_5551:
+					src = buf16[y * buf.GetStride() + x];
+					if (rev) {
+						src = bswap16(src);
+					}
+					r = Convert5To8((src >> 0) & 0x1F);
+					g = Convert5To8((src >> 5) & 0x1F);
+					b = Convert5To8((src >> 10) & 0x1F);
+					break;
+				case GPU_DBG_FORMAT_4444:
+					src = buf16[y * buf.GetStride() + x];
+					if (rev) {
+						src = bswap16(src);
+					}
+					r = Convert4To8((src >> 0) & 0xF);
+					g = Convert4To8((src >> 4) & 0xF);
+					b = Convert4To8((src >> 8) & 0xF);
+					break;
+				case GPU_DBG_FORMAT_8888:
+					src = buf32[y * buf.GetStride() + x];
+					if (rev) {
+						src = bswap32(src);
+					}
+					r = (src >> 0) & 0xFF;
+					g = (src >> 8) & 0xFF;
+					b = (src >> 16) & 0xFF;
+					break;
+				default:
+					ERROR_LOG(COMMON, "Unsupported framebuffer format for screenshot: %d", buf.GetFormat());
+					return nullptr;
+				}
+			}
+		}
+		buffer = temp;
+	}
+
+	return buffer;
+}
+
+void AVIDump::AddFrame()
+{
+	GPUDebugBuffer buf;
+	gpuDebug->GetCurrentFramebuffer(buf);
+	u32 w = buf.GetStride();
+	u32 h = buf.GetHeight();
+	CheckResolution(w, h);
+	u8 *flipbuffer = nullptr;
+	const u8 *buffer = ConvertBufferTo888RGB(buf, flipbuffer, w, h);
+	s_src_frame->data[0] = const_cast<u8*>(buffer);
+	s_src_frame->linesize[0] = w * 3;
+	s_src_frame->format = AV_PIX_FMT_RGB24;
+	s_src_frame->width = s_width;
+	s_src_frame->height = s_height;
+
+	// Convert image from BGR24 to desired pixel format, and scale to initial
+	// width and height
+	if ((s_sws_context =
+           sws_getCachedContext(s_sws_context, w, h, AV_PIX_FMT_RGB24, s_width, s_height,
+                                s_stream->codec->pix_fmt, SWS_BICUBIC, nullptr, nullptr, nullptr)))
+	{
+		sws_scale(s_sws_context, s_src_frame->data, s_src_frame->linesize, 0, h,
+              s_scaled_frame->data, s_scaled_frame->linesize);
```

Comment:
Please use tabs to indent.  We've included an editorconfig file that can automatically configure your indent styles per project:

https://github.com/hrydgard/ppsspp/blob/master/.editorconfig
http://editorconfig.org/

-[Unknown]